### PR TITLE
ACME: improve documentation

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Improves/fixes various aspects of documentation for the ACME modules:
 1. Changing some references to Let's Encrypt to be more generic (i.e. less vendor specific ;-) );
 1. Updating draft references to most current drafts;
 1. Using `notes:` instead of `description:` for everything which is more a note. This also fixes a problem with `description:` being used in the docs fragment causing the descriptions of the modules being sorted alphabetically (!);
 1. Harmonizing short description and first paragraph of description for all ACME modules;
 1. Adding more cross-references between modules, in particular to the new `acme_challenge_cert_helper` and `certificate_complete_chain` modules.

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
acme_account
acme_account_facts
acme_certificate
acme_certificate_revoke
acme_challenge_cert_helper
lib/ansible/module_utils/acme.py
lib/ansible/utils/module_docs_fragments/acme.py

##### ANSIBLE VERSION
```
2.7.0
```

### CODE REVIEW

File: lib/ansible/modules/crypto/acme/acme_certificate_revoke.py [hunk 1]
```diff
@@ -38,6 +38,29 @@
     description:
       - "Path to the certificate to revoke."
     required: yes
+  account_key_src:
```
Comment (resmo) [inline lib/ansible/modules/crypto/acme/acme_certificate_revoke.py:41]: oh, did we really miss these? and even the bot didn't notice?

**Review by resmo — APPROVED** [review]
shipit

File: lib/ansible/modules/crypto/acme/acme_certificate_revoke.py (hunk 1 above)
Comment (felixfontein) [reply lib/ansible/modules/crypto/acme/acme_certificate_revoke.py:41]: Nope, we didn't :) These two were in the doc fragment. The point I added them here as well is that here the description should be slighly different: as opposed to the other modules, where `account_key_src` and `account_key_content` are mutually exclusive, we have the 4-group `account_key_src`, `account_key_content`, `private_key_source`, `private_key_content` for this module. Before this override, the description for these two mentioned only these two, but the description for the others mentioned all four (as well as the general description).

**Comment by felixfontein** [comment]
Thanks for reviewing!

**Comment by felixfontein** [comment]
bot_status

**Pull request merged**